core/trie/src/lib.rs

```diff
@@ -152,16 +166,22 @@ pub fn child_trie_root<H: Hasher, I, A, B>(_storage_key: &[u8], input: I) -> Vec
 }
 
 /// Determine a child trie root given a hash DB and delta values. H is the default hasher, but a generic implementation may ignore this type parameter and use other hashers.
-pub fn child_delta_trie_root<H: Hasher, I, A, B>(_storage_key: &[u8], db: &mut HashDB<H>, root_vec: Vec<u8>, delta: I) -> Result<Vec<u8>, Box<TrieError<H::Out>>> where
+pub fn child_delta_trie_root<H: Hasher, I, A, B, DB>(
+	_storage_key: &[u8],
+	db: &mut DB,
+	root_vec: Vec<u8>,
+	delta: I
+) -> Result<Vec<u8>, Box<TrieError<H::Out>>> where
 	I: IntoIterator<Item = (A, Option<B>)>,
 	A: AsRef<[u8]> + Ord,
 	B: AsRef<[u8]>,
+	DB: hash_db::HashDB<H, trie_db::DBValue> + hash_db::PlainDB<H::Out, trie_db::DBValue>,
```

Comment:
This looks like can't use the `HashDB` type alias defined above because Rust treats type alias and traits differently.

Reply:
there has been an RFC for trait aliases for ages but not happening anytime soon sadly. type aliases of traits are actually for the unsized values with the type of the trait.